### PR TITLE
docs(contributing/coding-guidelines): add launch files documentation for ROS nodes guidelines

### PR DESCRIPTION
**Pull request by liuzf1988**
## Description

Add a brief introduction of the launch file organization in Autoware. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/contributing/coding-guidelines/ros-nodes/launch-files.md [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+
+### Create or add a new package in Autoware
+
+If a newly created package has executable node, we expect sample launch file and configuration within the package, just like the recommended structure shown in previous [directory structure](https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/directory-structure/) page.
+
+In order to automatically load the newly added package when starting Autoware, you may choose to call its launch file at certain place. For example, if using ICP instead of NDT as the pointcloud registration algorithm, you can modify the `tier4_localization_launch/launch/pose_estimator/pose_estimator.launch.xml` file to load the newly added ICP package.
+
+## Future updates
+
+How we organize configuration files.
```
Comment (mitsudome-r) [inline docs/contributing/coding-guidelines/ros-nodes/launch-files.md:47]: We can work on this once https://github.com/autowarefoundation/autoware_launch/issues/66 is closed

Comment (liuzf1988) [reply docs/contributing/coding-guidelines/ros-nodes/launch-files.md:47]: > We can work on this once [autowarefoundation/autoware_launch#66](https://github.com/autowarefoundation/autoware_launch/issues/66) is closed

Thanks! After the issue is finished, I will make some update.

File: docs/contributing/coding-guidelines/ros-nodes/launch-files.md [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+tier4_vehicle_launch    tier4_localization_launch   tier4_system_launch tier4_sensing_launch    tier4_perception_launch    ...
+                                |
+─────────────────────────────────────────────────────────────────
+|               |                   |                           |
+pose_estimator twist_estimator    pose_twist_fusion_filter     ...
+                                        |
+                    ─────────────────────────────────────────
+                    |                   |                   |
+                    ekf_localizer    stop_filter        twist2accel
+```
```
Comment (xmfcx) [inline docs/contributing/coding-guidelines/ros-nodes/launch-files.md:37]: Can you convert this into a mermaid diagram?
You can follow https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams to do it.

**Review by xmfcx — CHANGES_REQUESTED** [review]

**Comment by xmfcx** [comment]
Thanks for working on this PR,

I've converted it to draft, please mark it as "Ready for review" once it is ready again!

Screenshot from 2023-01-07 01-24-40

File: docs/contributing/coding-guidelines/ros-nodes/launch-files.md (hunk 2 above)
Comment (kenji-miyake) [reply docs/contributing/coding-guidelines/ros-nodes/launch-files.md:37]: Also, I'd like to add a note that says `tier4_*_launch` are just samples and anyone can propose their own launch packages.

**Comment by kenji-miyake** [comment]
FYI: For the semantic-pull-request failure, please refer to the documentation: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/ci-checks/#semantic-pull-request

**Comment by liuzf1988** [comment]
Sorry to forget to handle this PR in time. It has been updated

File: docs/contributing/coding-guidelines/ros-nodes/launch-files.md (hunk 2 above)
Comment (liuzf1988) [reply docs/contributing/coding-guidelines/ros-nodes/launch-files.md]: The pr has updated as required

File: docs/contributing/coding-guidelines/ros-nodes/launch-files.md [hunk 3]
```diff
@@ -1,5 +1,67 @@
 # Launch files
 
-!!! warning
+## Overview
 
-    Under Construction
+Autoware use ROS2 launch system to startup the software. Please see the [official documentation](https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Launch-Main.html) to get a basic understanding about ROS 2 Launch system if you are not familiar with it.
```
Comment (mitsudome-r) [inline docs/contributing/coding-guidelines/ros-nodes/launch-files.md:5]:
```suggestion
Autoware uses ROS 2 launch system to startup the software. Please see the [official documentation](https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Launch-Main.html) to get a basic understanding about ROS 2 Launch system if you are not familiar with it.
```

File: docs/contributing/coding-guidelines/ros-nodes/launch-files.md [hunk 4]
```diff
@@ -1,5 +1,67 @@
...
+
+## Guideline
+
+### The organization of launch files in Autoware
+
+Autoware mainly has two repositories related to launch file organization: the [autoware.universe](https://github.com/autowarefoundation/autoware.universe) and the [autoware_launch](https://github.com/autowarefoundation/autoware_launch).
+
+#### autoware.universe
+
+the `autoware.universe` contains the code of the main Autoware modules, and its `launch` directory is responsible for launching the nodes of each module. Autoware software stack is organized based on the [architecture](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-architecture/#high-level-architecture-design), so you may find that we try to match the launch structure similar to the architecture (splitting of files, namespace). For example, the `tier4_map_launch` subdirectory corresponds to the map module, so do the other `tier4_*_launch` subdirectories.
```
Comment (mitsudome-r) [inline docs/contributing/coding-guidelines/ros-nodes/launch-files.md:15]:
```suggestion
The `autoware.universe` contains the code of the main Autoware modules, and its `launch` directory is responsible for launching the nodes of each module. Autoware software stack is organized based on the [architecture](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-architecture/#high-level-architecture-design), so you may find that we try to match the launch structure similar to the architecture (splitting of files, namespace). For example, the `tier4_map_launch` subdirectory corresponds to the map module, so do the other `tier4_*_launch` subdirectories.
```

**Comment by liuzf1988** [comment]
> Thanks for working on this PR,
> 
> I've converted it to draft, please mark it as "Ready for review" once it is ready again!
> 
> Screenshot from 2023-01-07 01-24-40

Hi, @xmfcx , this PR has been ready for a while. And it requires your confirmation before merging

**Review by mitsudome-r — APPROVED** [review]

**Comment by liuzf1988** [comment]
@mitsudome-r Hi, this PR cannot be automatically merged, indicating the need for confirmation from xmfcx. I had  @ him before and have not replied

**Review by xmfcx — APPROVED** [review]

**Comment by xmfcx** [comment]
I'm sorry for taking long, it looks great!

**Pull request merged**